hearpreprocess/util/audio.py

```diff
@@ -14,17 +14,30 @@
 
 def mono_wav_and_fix_duration(in_file: str, out_file: str, duration: float):
     """
-    Convert to WAV file and trim to be equal to or less than a specific length
-
-    If the audio is already mono and has the required duration, the step is skipped
+    This 
+    1. Pads and trims the audio to the desired input duration
+    2. Converts to mono if more than 1 stream is present
+    3. Converts the audio to wav format
+    All the above are only done when the input file doesnot match the required
+    conditions of duration, streams and extension.
+    In case, we are unable to get any audio stats, by default all the steps are done.
+
+    If the audio is already satisfies all the expected filters, the step is skipped
     and a symlink is created to the original file
     """
     # Get the audio stats for the audio file
     audio_stats = get_audio_stats(in_file)
-    # Get the filters to apply to the audio
-    duration_filter_flag = audio_stats["duration"] != duration
-    mono_filter_flag = not audio_stats["mono"]
-    ext_filter_flag = audio_stats["ext"].lower() != ".wav"
+    # Get the filters to apply to the audio. 
+    # If ffmpeg probe is unable to fetch the audio stats
+    # in which case the stats will be empty
+    # we will make all the filters as True
+    if audio_stats:
+        duration_filter_flag = audio_stats["duration"] != duration
+        mono_filter_flag = not audio_stats["mono"]
```

Comment:
_flag is a terrible variable name because it doesn't tell you the orientation.

What about "_incorrect"?